tests/importer/importer.cpp

```diff
@@ -0,0 +1,420 @@
+
+    c->setImportReference("myComponent");
+    c->setImportSource(importSource);
+    model->addComponent(c);
+
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
+    model = importer->flatten(model);
+
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(0), validator->errorCount());
+
+    auto printer = libcellml::Printer::create();
+    EXPECT_EQ(e, printer->printModel(model));
+}
+
+TEST(Importer, duplicatesImportedOnceOnly)
+{
+    std::string in = "<?xml version=\"1.0\" encoding=\"iso-8859-1\"?>\n"
+                     "<model name=\"forked_import\" xmlns=\"http://www.cellml.org/cellml/2.0#\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns:xlink=\"http://www.w3.org/1999/xlink\">\n"
+                     "  <import xlink:href=\"prong.cellml\">\n"
+                     "    <component name=\"left\" component_ref=\"component1\" />\n"
+                     "  </import>\n"
+                     "  <import xlink:href=\"prong.cellml\">\n"
+                     "    <component name=\"right\" component_ref=\"component1\" />\n"
+                     "  </import>\n"
+                     "  <import xlink:href=\"prong.cellml\">\n"
+                     "    <component name=\"centre\" component_ref=\"component1\" />\n"
+                     "  </import>\n"
+                     "</model>";
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(in);
+    auto importer = libcellml::Importer::create();
+
+    importer->resolveImports(model, resourcePath("resolveimports/"));
+    EXPECT_EQ(size_t(3), model->componentCount()); // Three imported components ...
+    EXPECT_EQ(size_t(1), importer->libraryCount()); // ... but just one imported model.
+    EXPECT_FALSE(model->hasUnresolvedImports());
+}
+
+TEST(Importer, accessImportedModelLibrary)
+{
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("resolveimports/diamond.cellml"));
+    auto importer = libcellml::Importer::create();
+    auto printer = libcellml::Printer::create();
+
+    importer->resolveImports(model, resourcePath("resolveimports/"));
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
+    // Library should contain left, right, and one instance (not two) of the point.
+    EXPECT_EQ(size_t(3), importer->libraryCount());
+
+    auto left = importer->library(resourcePath("resolveimports/diamond_left.cellml"));
+    auto right = importer->library(resourcePath("resolveimports/diamond_right.cellml"));
+    auto point = importer->library(resourcePath("resolveimports/diamond_point.cellml"));
+
+    // Access library items by their URL.
+    EXPECT_EQ(fileContents("resolveimports/diamond_left.cellml"), printer->printModel(left));
+    EXPECT_EQ(fileContents("resolveimports/diamond_right.cellml"), printer->printModel(right));
+    EXPECT_EQ(fileContents("resolveimports/diamond_point.cellml"), printer->printModel(point));
+}
+
+TEST(Importer, multipleModelResolution)
+{
```

Comment:
@nickerso Is this the kind of functionality you meant?

Reply:
yep, in general :)